### ISSUE TITLE
[BUG] 'The attribute can't be saved' error

### ISSUE THREAD

**Issue by hostep** [tig-nl/postcode-magento2#43]
**To Reproduce**
Using:
- Magento OS 2.4.3-p2
- tig/postcode-magento2 1.5.1

1. Run `bin/magento setup:upgrade` after installing the module using composer

**Expected result**
No errors

**Actual result**
```
Unable to apply data patch TIG\Postcode\Setup\Patch\Data\AddTigEavAttributes for module TIG_Postcode. Original exception message: The attribute can't be saved.
```

After some quick debugging, this happens with the attribute `tig_housenumber` in the `addAttributeToForms` method the moment `$this->attributeRepository->save()` is called.

**Workaround**
None

**Screenshots**
None

**Please complete the following information**
- Magento version: OS 2.4.3-p2
- PHP version: 7.4.30

**Extra**
This is on a nearly empty Magento installation, only contains a couple of 3rd party modules.
If you are unable to reproduce, please let me know and I'll put some more time in trying to find better steps to reproduce

Thanks!

**Comment by tig-jeffreybranderhorst** [comment]
Hi @hostep ,

We have installed the extension on a clean Magento without errors. We were also able to run setup:upgrade without issues. I cannot reproduce your issue. If you are going to investigate this further but cannot find a reason why you are getting this error, you can contact us at support@tig.nl, so we can help you find the problem on your environment. 

Have a great day,
Jeffrey

**Comment by hostep** [comment]
Thanks @tig-jeffreybranderhorst

It's an extremely weird situation, when I look at the exception that's catched [over here](https://github.com/magento/magento2/blob/2.4.3-p2/app/code/Magento/Eav/Model/AttributeRepository.php#L90):
- message: `Attribute with the same code already exists.`
- trace:
```
#0 vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(416): Magento\Framework\Model\ResourceModel\Db\AbstractDb->_checkUnique()
#1 vendor/magento/module-eav/Model/AttributeRepository.php(89): Magento\Framework\Model\ResourceModel\Db\AbstractDb->save()
#2 vendor/tig/postcode-magento2/Setup/Patch/Data/AddTigEavAttributes.php(76): Magento\Eav\Model\AttributeRepository->save()
#3 vendor/tig/postcode-magento2/Setup/Patch/Data/AddTigEavAttributes.php(146): TIG\Postcode\Setup\Patch\Data\AddTigEavAttributes->addAttributeToForms()
#4 vendor/magento/framework/Setup/Patch/PatchApplier.php(162): TIG\Postcode\Setup\Patch\Data\AddTigEavAttributes->apply()
#5 setup/src/Magento/Setup/Model/Installer.php(1094): Magento\Framework\Setup\Patch\PatchApplier->applyDataPatch()
#6 setup/src/Magento/Setup/Model/Installer.php(960): Magento\Setup\Model\Installer->handleDBSchemaData()
#7 setup/src/Magento/Setup/Console/Command/UpgradeCommand.php(147): Magento\Setup\Model\Installer->installDataFixtures()
#8 vendor/symfony/console/Command/Command.php(255): Magento\Setup\Console\Command\UpgradeCommand->execute()
#9 vendor/symfony/console/Application.php(1009): Symfony\Component\Console\Command\Command->run()
#10 vendor/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#11 vendor/magento/framework/Console/Cli.php(115): Symfony\Component\Console\Application->doRun()
#12 vendor/symfony/console/Application.php(149): Magento\Framework\Console\Cli->doRun()
#13 bin/magento(23): Symfony\Component\Console\Application->run()
#14 {main}
```

And the attribute code at that time is `tig_housenumber`


But when I search the database, there is no attribute that exists with that code:

```
mysql> SELECT * FROM eav_attribute WHERE attribute_code LIKE 'tig_%';
Empty set (0.00 sec)
```

So if I'm theorising, it sounds like Magento is trying to generate the attribute twice in a transaction and then rolls back everything when it fails or something like that. Or another theory is that its cache thinks the attribute already exists without querying the database.


Very weird, I'll try to continue digging later today.

**Comment by hostep** [comment]
Okay, I found something, when I disable another third party module `hyva-themes/magento2-theme-module` (which comes as a dependency of the [hyva theme frontend framework](https://hyva.io/)), it starts working correctly.

So I'm going to dig into that module a bit and maybe ask around with the devs of that module.

Will close this issue now, since it doesn't seem to be a bug in this module.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @hostep ,

Thank you for all the updates! I will also speak about this internally, maybe we can also find something that helps for the combination of extensions. 

Thank you again for all the clear updates and explanations.

Have a great day,
Jeffrey

**Comment by hostep** [comment]
Got confirmation from the Hyvä team that it's a bug in their module and a fix already exists for it. I can confirm that it fixes the issue. The bug got introduced in version 1.1.15 of `hyva-themes/magento2-theme-module` and is planned to be fixed in version 1.1.17

Fix exists here (for the people who have access): https://gitlab.hyva.io/hyva-themes/magento2-theme-module/-/merge_requests/222/diffs

**Comment by tig-jeffreybranderhorst** [comment]
Hey @hostep ,

That is great to read, thank you for the update and good to read this is going to be fixed! 

Met vriendelijke groet / Kind regards / Mit freundlichen Grüßen / 最好的祝福,

Jeffrey Branderhorst
CUSTOMER CARE SPECIALIST
